cmd/node/node.go

```diff
@@ -618,7 +618,22 @@ func (app *App) initServices(ctx context.Context,
 	app.host.Register(beacon.FollowingVotesProtocol,
 		pubsub.ChainGossipHandler(syncHandler, beaconProtocol.HandleFollowingVotes))
 	app.host.Register(proposals.NewProposalProtocol, pubsub.ChainGossipHandler(syncHandler, proposalListener.HandleProposal))
-	app.host.Register(activation.AtxProtocol, pubsub.ChainGossipHandler(syncHandler, atxHandler.HandleGossipAtx))
+
+	// ValidatePost is CPU intensive. limit its number of workers.
+	numWorkers := runtime.NumCPU()
+	if numWorkers > 1 {
+		numWorkers /= 2
+	}
```

Comment:
i remember that it was discussed before, but we shouldn't do it. we have a fetcher codepath in every handler, which can take unpredictable amount of time

Reply:
my rationale is that once we enter gossip mode, fetcher should fetch minimal amount of ATXs in a steady state. (unless we are doing a network rejoin).

so when we start syncing, we spend most of the CPUs processing ATX and nothing else. once ATXs are synced, then the majority of ATX processing is on half of the CPUs.

in any case, i am fine removing the CPUs part from this PR and deal with it later when it becomes a problem.

Reply:
one interesting thing: with the 1/2 CPUs dedicated to ATX gossip via p2p validator, systest failed across the board. i looked briefly and see a lot more hare/beacon errors because miner has no ATXs. that seems to hint that the node wasn't able to process all the ATXs in time.

Reply:
they may have been dropped as well. i don't remember what exactly is controlled with a variable that you changed. and what happens on overflow